assets/src/dashboard/app/views/editorSettings/publisherLogo/index.js

```diff
@@ -66,39 +71,154 @@ function PublisherLogoSettings({
   publisherLogos,
   uploadError,
 }) {
+  const { isRTL } = useConfig();
+
+  const containerRef = useRef();
+  const gridRef = useRef();
+  const itemRefs = useRef({});
+
+  const [activePublisherLogo, _setActivePublisherLogoId] = useState(null);
+  const [indexRemoved, setIndexRemoved] = useState(null);
+
+  const publisherLogosById = useMemo(() => publisherLogos.map(({ id }) => id), [
+    publisherLogos,
+  ]);
+
+  const publisherLogoCount = useRef(publisherLogosById.length);
+
+  const onRemoveLogoClick = useCallback(
+    (e, { publisherLogo, idx }) => {
+      e.preventDefault();
+
+      handleRemoveLogo(publisherLogo);
+      setIndexRemoved(idx);
+      publisherLogoCount.current = publisherLogosById.length;
+    },
+    [handleRemoveLogo, publisherLogosById.length]
+  );
+
+  const setActivePublisherLogoId = useCallback((id) => {
+    _setActivePublisherLogoId(id);
+  }, []);
+
+  // set active logo when first painting
+  useEffect(() => {
+    if (publisherLogos.length > 0 && !activePublisherLogo) {
+      setActivePublisherLogoId(publisherLogos?.[0].id);
+    }
+  }, [activePublisherLogo, publisherLogos, setActivePublisherLogoId]);
+
+  // Update publisher logo focus when logo is removed
+  useEffect(() => {
+    if (
+      Boolean(indexRemoved?.toString()) &&
+      publisherLogosById.length !== publisherLogoCount.current
+    ) {
+      if (publisherLogosById.length === 0) {
+        // if the user has removed their last publisher logo, the logo grid will not render
+        // the first element child of containerRef becomes the input upload
+        // upload will always be present unless upload is not enabled.
+        // currently only admin can get to settings, which means they have upload ability
+        // checking for current and firstElementChild are safeguards
+        return containerRef.current?.firstElementChild
+          ?.getElementsByTagName('input')[0]
+          .focus();
```

Comment:
We could add a ref to the input and focus on that instead of all of the optional chaining.

Reply:
the file uploader is using refs already that i don't want to add to, i did however realize that this is nesting unnecessarily. so i cleaned it up a bit: `containerRef.current?.getElementsByTagName('input')[0].focus();` which feels a lot better.